app/services/setting_getter.rb

```diff
@@ -1,17 +1,29 @@
 # frozen_string_literal: true
 
 class SettingGetter
-  def initialize(setting_name:, provider:)
+  include Rails.application.routes.url_helpers
+
+  def initialize(setting_name:, provider:, host: nil)
     @setting_name = setting_name
     @provider = provider
+
+    return if host.nil? || setting_name != 'BrandingImage'
+
+    Rails.application.routes.default_url_options[:host] = host # Only needed using image attachment
```

Comment:
I'd remove this line and instead in an initializer I'd set `Rails.application.routes.default_url_options[:only_path] = true` this will give the expected outcome without introducing an error surface.